### ISSUE TITLE
"java.lang.IndexOutOfBoundsException" when connecting to 23.88.73.88:14594

### ISSUE THREAD

**Issue by swooboo** [Anuken/Mindustry#9279]
### Platforms

Windows

### Build

146

### Issue

When connecting to `23.88.73.88:14594` (The Quantum Bell), getting the error on the screenshot. Same result with several setups:

1. Windows
2. Ubuntu (clean, new install)
3. WSL Ubuntu (clean, new install)
4. Android on home WiFi and also on mobile data (clean, new install)
5. Dev build `Mindustry-BE-Desktop-24571`, after changing the `version.properties` version to 146
6. Foo's client under Windows

One user in `#help` of the Mindustry Discord server reported that he could connect to it from Linux so I don't know what to think of it. Looks like the server reports that it's on 146.

![image](https://github.com/Anuken/Mindustry/assets/900603/34b52f7c-f471-4a93-9a6a-e631661a64db)
[crash_out_of_bounds.txt](https://github.com/Anuken/Mindustry/files/13350925/crash_out_of_bounds.txt)
[last_log.txt](https://github.com/Anuken/Mindustry/files/13350926/last_log.txt)

### Steps to reproduce

1. Run Mindustry client
2. Add the server `23.88.73.88:14594` to the server list
3. Connect to the server
4. Get `java.lang.IndexOutOfBoundsException` as described

### Mods used

Tried with Vanilla and no mods, then with Foo's client too.

### Save file

[game_data_out_of_bounds.zip](https://github.com/Anuken/Mindustry/files/13351329/game_data_out_of_bounds.zip)

### (Crash) logs

[last_log.txt](https://github.com/Anuken/Mindustry/files/13351332/last_log.txt)
[crash_out_of_bounds.txt](https://github.com/Anuken/Mindustry/files/13351333/crash_out_of_bounds.txt)

### Submission

- [X] I have updated to the latest release (https://github.com/Anuken/Mindustry/releases) to make sure my issue has not been fixed.
- [X] I have searched the closed and open issues to make sure that this problem has not already been reported.

**Comment by Anuken** [comment]
> Dev build Mindustry-BE-Desktop-24571, after changing the version.properties version to 146

You can't just change the version of a BE build and expect it to be network compatible with 146. That's not how it works. Only build 146 is actually compatible with 146 servers. Don't spoof versions.

**Comment by swooboo** [comment]
Wait. I used the vanilla client and it reproduces too. All the logs and the screenshot is from vanilla without mods and not a custom build. Should it really be closed?

**Comment by Anuken** [comment]
Yes. The server doesn't appear to be running vanilla; it's probably using some misconfigured mods that aren't included in the mod list send to clients. Either that, or it's not running 146 either. Ask the host.

**Comment by Beginnerexe** [comment]
Can confirm, the server is modded with 1 mod and is currently running v7 bulid 146.
The server is meant to be private but people really like leaking it for some reason.
